mythril/analysis/modules/ether_thief.py

```diff
@@ -74,54 +69,40 @@ def _analyze_state(state):
         :param state:
         :return:
         """
+        state = copy(state)
         instruction = state.get_current_instruction()
 
-        if instruction["opcode"] != "CALL":
-            return []
-
-        address = instruction["address"]
-
         value = state.mstate.stack[-3]
         target = state.mstate.stack[-2]
 
-        eth_sent_by_attacker = symbol_factory.BitVecVal(0, 256)
-
         constraints = copy(state.mstate.constraints)
 
-        for tx in state.world_state.transaction_sequence:
-            """
-            Constraint: The call value must be greater than the sum of Ether sent by the attacker over all
-            transactions. This prevents false positives caused by legitimate refund functions.
-            Also constrain the addition from overflowing (otherwise the solver produces solutions with 
-            ridiculously high call values).
-            """
-            constraints += [BVAddNoOverflow(eth_sent_by_attacker, tx.call_value, False)]
-            eth_sent_by_attacker = Sum(
-                eth_sent_by_attacker,
-                tx.call_value * If(tx.caller == ATTACKER_ADDRESS, 1, 0),
-            )
-
-            """
-            Constraint: All transactions must originate from regular users (not the creator/owner).
-            This prevents false positives where the owner willingly transfers ownership to another address.
-            """
-
-            if not isinstance(tx, ContractCreationTransaction):
-                constraints += [tx.caller != CREATOR_ADDRESS]
-
         """
```

Comment:
Yes, but then we would need to add some annotations to track the address of the call instructions, and it might be difficult to know which call instruction is the one we want to report. Any ideas on that front?

Reply:
One way might be we can probably iterate all the calls in the annotation and then add those as issues if each one of them singularly can create an ether theft.

Reply:
What about suicides? Do those also fall under "ether thief", or should we skip those? (Hooking STOP will detect suicides as well)

Reply:
I'd rather merge this in now, and then come back to improving the ether thief module in a different pull request. I considered hooking STOP, but it requires a more logic to be added, so a dedicated PR might make things easier.

Reply:
SUICIDE can also be placed in the terminal instruction set right?

Reply:
> One way might be we can probably iterate all the calls in the annotation and then add those as issues if each one of them singularly can create an ether theft.

That's how it's done in the integer module.

> What about suicides? Do those also fall under "ether thief", or should we skip those? 

They're treated as a separate module because we also count suicides that don't send Ether to the attacker like `selfdestruct(fixed_address)`.